ladybug/_datacollectionbase.py

```diff
@@ -256,8 +276,10 @@ def filter_by_pattern(self, pattern):
             A new Data Collection with filtered data
         """
         _filt_values, _filt_datetimes = self._filter_by_pattern(pattern)
-        collection = self.__class__(
-            self.header.duplicate(), _filt_values, _filt_datetimes)
+        if self._enumeration is None:
```

Comment:
why are we returning a mutable datacollection even if the original datacollection is immutable? shouldn't we just follow what the original object is?

Reply:
That was my original thought but then I realized what a mess this creates for all of the other methods that return data collections derived from a starting data collection.  For example, for all of the methods that return Daily, Monthly or MonthlyPerHour collections from hourly collections, do all of those also need to follow the same mutability as the starting collection?  If so, there will have to be code like the one you are commenting on here at least 20 times in datacollection.py since we will have to look up the mutable/immutable equivalent for Daily, Monthly or MonthlyPerHour collections on the hourly collection.

Then I asked myself, are there any cases I could think of where it is important that this mutability is preserved as one derives several new collections from a starting data collection. I could not think of any.  In fact, I would usually prefer to get a mutable version since it seems that the only cases where the immutable collections are important are the when they are tied to specific objects like the comfort collections.  So that is how I came up with these (hopefully pretty simple) set of rules that govern how immutability is conserved as one derives data collections from a starting collection:
https://github.com/chriswmackey/ladybug/blob/development/ladybug/datacollectionimmutable.py#L4-L12

Reply:
I think we both agree that this would be the expected behavior and I agree with you that it doesn't worth the effort at this point. Since it's well documented let's leave it as is and if someone wants to get started with contributing to the library this is a good one to take care of. We can always wrap this part of code in a utility function so we don't duplicate all the code. Thank you for the explanation.